autoendpoint/src/routers/apns/router.rs

```diff
@@ -136,16 +138,30 @@ impl ApnsRouter {
                     return ApiError::from(ApnsError::Unregistered);
                 } else {
                     warn!("APNS error: {:?}", response.error);
-                    incr_error_metric(&self.metrics, "apns", channel, "server_error");
+                    incr_error_metric(&self.metrics, "apns", channel, "server_error", code, None);
```

Comment:
I'm not sure what value `code` could become here but it looks like this `a2::Error` gets converted to an `ApnsError::ApnsUpstream` below making its code become a `BAD_GATEWAY` regardless.